src/System.IO/tests/Stream/Stream.ReadOnly.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // This class emulates the behavior of Stream.Null,
+    // except CanWrite == false and attempting to write
+    // will throw a InvalidOperationException.
+    public sealed class ReadOnlyStream : Stream
+    {
+        public override bool CanRead => true;
+        public override bool CanSeek => true;
+        public override bool CanWrite => false;
+        public override long Length => 0;
+
+        public override long Position
+        {
+            get { return 0; }
+            set { }
+        }
+
+        public override void Flush() { }
+        public override long Seek(long offset, SeekOrigin origin) => 0;
+        public override void SetLength(long value) { }
+        public override int Read(byte[] buffer, int offset, int count) => 0;
+        public override void Write(byte[] buffer, int offset, int count) { throw new InvalidOperationException(); }
```

Comment:
Didn't know how to quite work this out, since the check this was meant to validate also throws the same exception:

```cs
if (!destination.CanWrite)  
{  
    throw new NotSupportedException(SR.NotSupported_UnwritableStream);  
}
```

So then you wouldn't be able to tell whether the check was actually working, or it was getting thrown from `ReadOnlyStream.Write`. Perhaps I should just noop here instead?